src/stream_base.cc

```diff
@@ -303,17 +304,22 @@ int StreamBase::WriteString(const FunctionCallbackInfo<Value>& args) {
 }
 
 
-void StreamBase::CallJSOnreadMethod(ssize_t nread, Local<Object> buf) {
+void StreamBase::CallJSOnreadMethod(ssize_t nread,
+                                    Local<ArrayBuffer> ab,
+                                    size_t offset) {
   Environment* env = env_;
 
+#ifdef DEBUG
+  CHECK_EQ(static_cast<int32_t>(nread), nread);
+  CHECK_EQ(static_cast<int32_t>(offset), offset);
```

Comment:
IIUC this will also allow us to remove
`// TODO(bnoordhuis) Check that nread > 0.`
https://github.com/nodejs/node/pull/23797/files#diff-f45eb699237c2e38dc9b49b588933c11R497.
in `channel.onread`

Reply:
I’m not sure we can guarantee `nread > 0`, but `nread >= 0` should make sense.